[LTC] Refactor class Shape

alanwaketan opened the pull request:
Summary:
This pull request keeps track of refactoring of class Shape.
1. Remove Shape.IsTuple() from ShapeUtil.
2. Change Shape.at_element_type_ to Shape.scalar_type_.
3. Change Shape.dimensions_ to Shape.sizes_.
3. Change Shape.rank() to Shape.dim().
4. Change TsNode.shape_ to TsNode.shapes_.
5. Remove tuple shapes in class Shape.
4. Remove Shape.IsTuple().
5. A little bit more clean up on class Shape.

Test Plan:
lazy_tensor_core/test/cpp/build/test_ptltc

To be noted:
It's easier to review the pull request one commit by another.

Depends on: #67755.

Commit:
[LTC] Remove Shape.IsTuple() from ShapeUtil

Summary:
This commit removes Shape.IsTuple() from ShapeUtil as the first step to
get rid of tuple shapes within Shape.

Test Plan:
lazy_tensor_core/test/cpp/build/test_ptltc

Commit:
[LTC] Change Shape.at_element_type_ to Shape.scalar_type_

Summary:
This commit changes Shape.at_element_type_ to Shape.scalar_type_ as one
step to refactor class Shape.

Test Plan:
lazy_tensor_core/test/cpp/build/test_ptltc

Commit:
[LTC] Change Shape.dimensions_ to Shape.sizes_

Summary:
This commit changes Shape.dimensions_ to Shape.sizes_ as one step to
refactor class Shape.

Test Plan:
lazy_tensor_core/test/cpp/build/test_ptltc

Commit:
[LTC] Change Shape.rank() to Shape.dim()

Summary:
This commit changes Shape.rank() to Shape.dim() as one step to refactor
class Shape.

Test Plan:
lazy_tensor_core/test/cpp/build/test_ptltc

Commit:
[LTC] Change TsNode.shape_ to TsNode.shapes_

Summary:
This commit changes TsNode.shape_ to TsNode.shapes_ and remove all the
uses of tuple shapes.

Test Plan:
lazy_tensor_core/test/cpp/build/test_ptltc

Commit:
[LTC] Remove tuple shapes in class Shape

Summary:
This commit removes tuple shapes in class Shape.

Test Plan:
lazy_tensor_core/test/cpp/build/test_ptltc

Commit:
[LTC] Remove Shape.IsTuple()

Summary:
This commit removes Shape.IsTuple().

Test Plan:
lazy_tensor_core/test/cpp/build/test_ptltc

Commit:
[LTC] A little bit more clean up on class Shape

Summary:
This commit improves class Shape a bit.

Test Plan:
lazy_tensor_core/test/cpp/build/test_ptltc

wconstab commented on aten/src/ATen/templates/LazyIr.h:
```diff
@@ -10,22 +10,17 @@
 namespace ir {
 
-static lazy_tensors::Shape convertShape(
+static std::vector<lazy_tensors::Shape> convertShape(
```
convertShapes? (s)

alanwaketan replied:
Fixed.

wconstab commented on lazy_tensor_core/lazy_tensor_core/csrc/ops/cat.cpp:
```diff
@@ -7,22 +7,17 @@ namespace torch_lazy_tensors {
 
 // TODO(whc) don't duplicate this
-static lazy_tensors::Shape convertShape(
+static std::vector<lazy_tensors::Shape> convertShape(
```
should this be be moved into shape.h?

alanwaketan replied:
Yea, moved.

wconstab reviewed (approved):
Nice work, this is great stuff. Thanks!

Commit:
[LTC] Move convertShapes() to Shape.h

Summary:
This commit moves convertShapes() to Shape.h.

Test Plan:
lazy_tensor_core/test/cpp/build/test_ptltc

alanwaketan commented:
@wconstab Thanks for approving this pull request.
